satellite/repair/checker/checker.go

```diff
@@ -243,11 +283,26 @@ func (checker *Checker) updateSegmentStatus(ctx context.Context, pointer *pb.Poi
 		}
 
 		// add the entry if new or update attempt count if already exists
-		err := checker.irrdb.IncrementRepairAttempts(ctx, segmentInfo)
+		err := obs.irrdb.IncrementRepairAttempts(ctx, segmentInfo)
 		if err != nil {
-			return Error.New("error handling irreparable segment to queue %s", err)
+			obs.log.Sugar().Errorf("error handling irreparable segment to queue %s", err)
+			return nil
 		}
 	}
+
+	return nil
+}
+
+func (obs *checkerObserver) RemoteObject(ctx context.Context, path storj.Path, pointer *pb.Pointer) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	obs.monStats.remoteFilesChecked++
```

Comment:
How this metric works.

Some segments can be an object but not all of them, so if we repair them and they are objects then this method is called and we increment the metric, is that correct?

Reply:
Yes, if a remote segment is an object, it will be passed into both `RemoteObject` and `RemoteSegment`. Since we already do the checker things in `RemoteSegment`, we just keep track of one of the metrics to send datascience in `RemoteObject`. Before we were trying to calculate it ourselves from this file, but now it can be done more easily with the metainfo loop.